[dnf5] Improve sanitizers again

dmach opened the pull request.

Commit:
build: Rework sanitizers again, use LD_PRELOAD in the tests

Commit:
tests: Mark IdQueueTest::test_iterator_performance() as a performance test

This avoids allocating huge amount of memory when sanitizers are enabled

kontura commented:
I had to manually create `lsan-suppresions.txt`, we might want to commit it even if empty?

Also I see the `LD_PRELOAD` is used for python tests but for me the other (ruby, perl) tests fail without it as well with:
```
==158807==ASan runtime does not come first in initial library list; you should either link runtime to your application or manually preload it with LD_PRELOAD.
```

dmach commented:
Thanks for identifying the missing file.
I'll look into the other failing tests as well.

Commit:
tests: Fix memory leak in test_solv_query.t

Apply the same fix to Python and Ruby tests,
it makes the code more straightforward.

dmach commented:
@kontura new commits force-pushed, could you try once more?

kontura commented:
Thanks, LGTM.
